src/osrm_nb.cpp

```diff
@@ -1,22 +1,46 @@
 #include "osrm/osrm.hpp"
 
 #include "osrm/engine_config.hpp"
+#include "osrm/route_parameters.hpp"
+
+#include "utility/osrm_kwargs.h"
 
 #include <nanobind/nanobind.h>
+#include <nanobind/stl/string.h>
 
 namespace nb = nanobind;
+using namespace nb::literals;
 
 void init_EngineConfig(nb::module_& m);
+void init_BaseParameters(nb::module_& m);
+void init_RouteParameters(nb::module_& m);
 
 NB_MODULE(osrm_ext, m) {
-    namespace json = osrm::util::json;
     namespace api = osrm::engine::api;
+    namespace json = osrm::util::json;
 
     using osrm::OSRM;
     using osrm::engine::EngineConfig;
+    using osrm::engine::api::RouteParameters;
 
     init_EngineConfig(m);
+    init_BaseParameters(m);
+    init_RouteParameters(m);
 
     nb::class_<OSRM>(m, "OSRM", nb::is_final())
-        .def(nb::init<EngineConfig&>());    
+        .def(nb::init<EngineConfig&>())
+        .def("__init__", [](OSRM* t, const std::string& storage_path) { 
+            EngineConfig config;
+            config.storage_config = osrm::storage::StorageConfig(storage_path);
+
+            new (t) OSRM(config);
```

Comment:
Is "placement new" a recommended way of implementing constructors in nanobind?

Reply:
It appears that this is recommended because the memory was already allocated.

This behavior is referenced [here](https://nanobind.readthedocs.io/en/latest/porting.html#custom-constructors).
> The provided lambda function will be called with a pointer to uninitialized memory that has already been allocated (this memory region is co-located with the Python object for reasons of efficiency).